src/tools/auth0/handlers/scimHandler.ts

```diff
@@ -55,19 +56,16 @@ export default class ScimHandler {
 
 		for (let connection of connections) {
 			if (!this.isScimStrategy(connection.strategy)) continue;
-			
-			try {
-				this.idMap.set(connection.id, { strategy: connection.strategy, hasConfig: false });
-				await this.getScimConfiguration({ id: connection.id });
-				this.idMap.set(connection.id, { ...this.idMap.get(connection.id)!, hasConfig: true });
-
-				// To avoid rate limiter error, we making API requests with a small delay.
-				// TODO: However, this logic needs to be re-worked.
-				await sleep(500);
-			} catch (err) {
-				// Skip the connection if it returns 404. This can happen if `SCIM` is not enabled on a `SCIM` connection. 
-				if (err !== 'SCIM_NOT_FOUND') throw err;
-			}
+      
+      // To avoid rate limiter error, we making API requests with a small delay.
+      // TODO: However, this logic needs to be re-worked.
+      await sleep(500);
```

Comment:
Why is the SCIM update hitting rate limits when other connection updates are not?

Reply:
Hello @bweisberger,

The main challenge we're facing is that there isn't a straightforward way to get a complete list of SCIM-enabled connections. As a result, we have to manually check each SCIM connection to see if it's enabled or not, which is a limitation of the API.

While going through this process, we encountered rate limits (429 errors) when using the `/v2/connections/get-scim-configuration` endpoint, which led us to introduce `sleep` in the operations.

Another complication is that `auth0-deploy-cli` primarily relies on `node-auth0` for its features. However, the current version of `node-auth0` used in `auth0-deploy-cli` doesn't support SCIM. To work around this, we're directly using `Management API` endpoints to handle SCIM functionalities.

We understand this isn't the most efficient solution, but it's the approach we've had to take for now.

Reply:
Got it, makes sense. I think we're going to stay pinned to the version before 7.23.0 this one for now - we don't use SCIM and it looks like this change would add at least a couple minutes to our deploys. I'll be following along though as you all work through solutions. Thanks!

Reply:
@bweisberger,
Got it! Yes, SCIM was the only feature introduced in version 7.23.0. If you're not utilizing SCIM functionality, sticking with the previous version makes sense, especially if it helps keep your deploy times shorter by a few minutes.

Thank you !!